ethers-solc/src/artifacts/mod.rs

```diff
@@ -1365,19 +1365,33 @@ pub struct UserDoc {
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub kind: Option<String>,
     #[serde(default, skip_serializing_if = "::std::collections::BTreeMap::is_empty")]
-    pub methods: BTreeMap<String, UserDocNotice>,
+    pub methods: BTreeMap<String, UserDocMethodNotice>,
+    #[serde(default, skip_serializing_if = "::std::collections::BTreeMap::is_empty")]
+    pub events: BTreeMap<String, UserDocEventNotice>,
+    #[serde(default, skip_serializing_if = "::std::collections::BTreeMap::is_empty")]
+    pub errors: BTreeMap<String, Vec<UserDocErrorNotice>>,
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub notice: Option<String>,
 }
 
 #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
 #[serde(untagged)]
-pub enum UserDocNotice {
+pub enum UserDocMethodNotice {
     // NOTE: this a variant used for constructors on older solc versions
     Constructor(String),
     Method { notice: String },
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct UserDocEventNotice {
+    pub notice: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct UserDocErrorNotice {
+    pub notice: String,
+}
```

Comment:
can we unify this to `UserDocNotice`?

Reply:
then i think we need to rename `UserDocNotice::Method` to something like `UserDocNotice::Notice`